### PR TITLE
[Titanic] added launch_check_training to see if the model can check an epoch

### PR DESCRIPTION
**Pull request by mathysgrapotte**
Left TODO: 

- [ ] Add check_training as a nextflow module to be run at the top of the pipeline
- [x] Update launch_training so that it can take .yaml config as defined in the input 
- [ ] Check end-to-end nextflow pipeline if it can train for titanic 
- [ ] Create a test profile that runs models for titanic instead of the dummy one currently implemented

### CODE REVIEW

File: bin/launch_check_model.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            if torch.equal(initial_model_weights[key], self.model.state_dict()[key]):
+                print(f"Model weights have not changed for key {key}")
+            else:
+                print(f"Model weights have changed for key {key}")
+
+        # print the computed loss, displaying loss_fn name as well
+        print(f"Loss computed with {self.loss_fn.__class__.__name__} is {loss.item()}")
+
+
+def arg_parser():
```
Comment (alessiovignoli) [inline bin/launch_check_model.py:42]: can you move this block of lines after the imports? is better for consistency with the other launchers.

File: bin/tests/test_data/titanic/process_titanic_to_stimulus.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        "Parch",
+        "Fare",
+        "Embarked"
+    ])
+
+    df = df.drop_nulls()
+
+    # Rename columns to match stimulus format
+
+    df = df.rename({"Survived": "survived:label:int_class",
```
Comment (suzannejin) [inline bin/tests/test_data/titanic/process_titanic_to_stimulus.py:29]: why you have int_class, int_reg, but only float?
Also what is the difference between str_class and str?

File: bin/tests/test_model/titanic_model.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        x = self.relu(self.input_layer(x))
+        for layer in self.intermediate:
+            x = self.relu(layer(x))
+        x = self.softmax(self.output_layer(x))
+        return x
+    
+    def compute_loss(self, loss_fn, output, survived):
+        return loss_fn(output, survived)
+    
+    def epoch(self, x, y, loss_fn, optimizer):
```
Comment (suzannejin) [inline bin/tests/test_model/titanic_model.py:25]: also loss_fn should be a dict, isn't it? 
So that for each target y, it computes certain loss, according to the config

**Review by alessiovignoli — CHANGES_REQUESTED** [review]
address comments

File: bin/tests/test_data/titanic/process_titanic_to_stimulus.py (hunk 2 above)
Comment (mathysgrapotte) [reply bin/tests/test_data/titanic/process_titanic_to_stimulus.py:29]: This is to illustrate how someone could implement different "type names", int_class for instance gets binarized, and int_reg gets transformed into ranks (see TitanicExperiment in experiments.py for example)

File: bin/tests/test_model/titanic_model.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mathysgrapotte) [reply bin/tests/test_model/titanic_model.py:25]: I am thinking that the unit of information is an epoch (actually it is a batch, maybe it should be called batch instead). Then the tuner calls however many batches are necessary.

Comment (mathysgrapotte) [reply bin/tests/test_model/titanic_model.py:25]: loss_fn is a argument, name should match the losses defined in the config with the losses defined in the "epoch" method, thus, if the user has to apply various loss_fn, user could define multiple losses with the appropriate names in the epoch definition.

**Review by alessiovignoli — COMMENTED** [review]
very cool, approved so far

**Comment by mathysgrapotte** [comment]
The pipeline should now be ready to run titanic correctly on the python side. 
Next task should be to add nextflow capability for doing so and running titanic end-to-end on the nextflow side.

File: bin/tests/test_data/titanic/process_titanic_to_stimulus.py
[the remark this answers is not in the document]
Comment (mathysgrapotte) [reply bin/tests/test_data/titanic/process_titanic_to_stimulus.py:1]: Will be moved once we have the script that downloads titanic data from kaggle

File: bin/tests/test_model/simple_config.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
+model_params:
+  kernel_size:
+    space: [3]
+    mode: choice
+  pool_size:
+    space: [2]
+    mode: choice
+
+optimizer:
```
Comment (suzannejin) [inline bin/tests/test_model/simple_config.yaml:9]: optimizer can be merged to optimizer_params

**Review by suzannejin — APPROVED** [review]

**Review by alessiovignoli — APPROVED** [review]
